src/bus.rs

```diff
@@ -244,6 +247,40 @@ impl<B: UsbBus> UsbBusAllocator<B> {
             .expect("alloc_ep failed")
     }
 
+    /// Allocates an isochronous endpoint.
+    ///
+    /// # Arguments
+    ///
+    /// * `synchronization` - Type of synchronization used by the endpoint
+    /// * `usage` - Whether the endpoint is data, explicit feedback, or data+implicit feedback
+    /// * `payload_size` - Payload size in bytes.
+    /// * `interval` - Interval for polling, expressed in frames/microframes.  Must be 1 to 16,
+    ///    calculated as 2^(interval-1), see USB 2.0 section 9.6.6.
+    ///
+    /// # Panics
+    ///
+    /// Panics if endpoint allocation fails, because running out of endpoints or memory is not
+    /// feasibly recoverable.
+    #[inline]
+    pub fn isochronous<D: EndpointDirection>(
+        &self,
+        synchronization: IsochronousSynchronizationType,
+        usage: IsochronousUsageType,
+        payload_size: u16,
+        interval: u8,
```

Comment:
Can we do input validation on the `interval` and `payload_size` to verify they conform with USB requirements?

Reply:
Ah, duh, I was thinking about https://github.com/rust-embedded-community/usb-device/issues/71

Reply:
Had a closer look at this tonight, I don't think we have quite enough information to validate at this level.  Might be able to do some of the fields, but others need to know about the USB version and speed of the device.  As in #71 I do think it would be appropriate to do in this crate, just a bit outside the scope of this PR.